diff -ru xerces-c-3.1.1/src/xercesc/dom/DOMImplementationList.hpp xerces-c-3.1.1-gpdb/src/xercesc/dom/DOMImplementationList.hpp
--- xerces-c-3.1.1/src/xercesc/dom/DOMImplementationList.hpp	2008-06-26 06:29:21.000000000 -0700
+++ xerces-c-3.1.1-gpdb/src/xercesc/dom/DOMImplementationList.hpp	2015-11-30 12:00:29.137035633 -0800
@@ -37,7 +37,7 @@
  * an integral index, starting from 0.
  */
 
-class  CDOM_EXPORT DOMImplementationList {
+class  CDOM_EXPORT DOMImplementationList : public XMemory {
 protected:
     // -----------------------------------------------------------------------
     //  Hidden constructors
diff -ru xerces-c-3.1.1/src/xercesc/dom/DOMNodeList.hpp xerces-c-3.1.1-gpdb/src/xercesc/dom/DOMNodeList.hpp
--- xerces-c-3.1.1/src/xercesc/dom/DOMNodeList.hpp	2008-06-26 06:29:21.000000000 -0700
+++ xerces-c-3.1.1-gpdb/src/xercesc/dom/DOMNodeList.hpp	2015-11-30 12:00:29.140035611 -0800
@@ -23,6 +23,7 @@
 #define XERCESC_INCLUDE_GUARD_DOMNODELIST_HPP
 
 #include <xercesc/util/XercesDefs.hpp>
+#include <xercesc/util/XMemory.hpp>
 
 XERCES_CPP_NAMESPACE_BEGIN
 
@@ -42,7 +43,8 @@
  * reflected in any DOMNodeLists that may have been created for that tree.
  */
 
-class  CDOM_EXPORT DOMNodeList {
+class  CDOM_EXPORT DOMNodeList : public XMemory 
+{
 protected:
     // -----------------------------------------------------------------------
     //  Hidden constructors
diff -ru xerces-c-3.1.1/src/xercesc/dom/impl/DOMDocumentImpl.cpp xerces-c-3.1.1-gpdb/src/xercesc/dom/impl/DOMDocumentImpl.cpp
--- xerces-c-3.1.1/src/xercesc/dom/impl/DOMDocumentImpl.cpp	2010-04-11 10:40:33.000000000 -0700
+++ xerces-c-3.1.1-gpdb/src/xercesc/dom/impl/DOMDocumentImpl.cpp	2015-11-30 12:00:29.145035574 -0800
@@ -312,7 +312,8 @@
         throw DOMException(
         DOMException::INVALID_CHARACTER_ERR, 0, getMemoryManager());
 
-    return new (this, DOMMemoryManager::DOCUMENT_TYPE_OBJECT) DOMDocumentTypeImpl(this, nam, false);
+	// avoid confusion with the XMemory member new operator
+    return ::new (this, DOMMemoryManager::DOCUMENT_TYPE_OBJECT) DOMDocumentTypeImpl(this, nam, false);
 }
 
 
@@ -326,7 +327,8 @@
         throw DOMException(
         DOMException::INVALID_CHARACTER_ERR, 0, getMemoryManager());
 
-    return new (this, DOMMemoryManager::DOCUMENT_TYPE_OBJECT) DOMDocumentTypeImpl(this, qualifiedName, publicId, systemId, false);
+	// avoid confusion with the XMemory member new operator
+    return ::new (this, DOMMemoryManager::DOCUMENT_TYPE_OBJECT) DOMDocumentTypeImpl(this, qualifiedName, publicId, systemId, false);
 }
 
 
@@ -916,7 +918,7 @@
 
     DOMDeepNodeListImpl* retList = fNodeListPool->getByKey(rootNode, tagName, 0);
     if (!retList) {
-        XMLSize_t id = fNodeListPool->put((void*) rootNode, (XMLCh*) tagName, 0, new (this) DOMDeepNodeListImpl(rootNode, tagName));
+        XMLSize_t id = fNodeListPool->put((void*) rootNode, (XMLCh*) tagName, 0, ::new (this) DOMDeepNodeListImpl(rootNode, tagName));
         retList = fNodeListPool->getById(id);
     }
 
@@ -935,7 +937,7 @@
     DOMDeepNodeListImpl* retList = fNodeListPool->getByKey(rootNode, localName, namespaceURI);
     if (!retList) {
         // the pool will adopt the DOMDeepNodeListImpl
-        XMLSize_t id = fNodeListPool->put((void*) rootNode, (XMLCh*) localName, (XMLCh*) namespaceURI, new (this) DOMDeepNodeListImpl(rootNode, namespaceURI, localName));
+        XMLSize_t id = fNodeListPool->put((void*) rootNode, (XMLCh*) localName, (XMLCh*) namespaceURI, ::new (this) DOMDeepNodeListImpl(rootNode, namespaceURI, localName));
         retList = fNodeListPool->getById(id);
     }
 
diff -ru xerces-c-3.1.1/src/xercesc/dom/impl/DOMDocumentTypeImpl.cpp xerces-c-3.1.1-gpdb/src/xercesc/dom/impl/DOMDocumentTypeImpl.cpp
--- xerces-c-3.1.1/src/xercesc/dom/impl/DOMDocumentTypeImpl.cpp	2008-07-22 03:56:56.000000000 -0700
+++ xerces-c-3.1.1-gpdb/src/xercesc/dom/impl/DOMDocumentTypeImpl.cpp	2015-11-30 12:00:29.146035567 -0800
@@ -208,11 +208,11 @@
     DOMNode* newNode = 0;
     DOMDocument* doc = castToNodeImpl(this)->getOwnerDocument();
     if (doc != 0)
-        newNode = new (doc, DOMMemoryManager::DOCUMENT_TYPE_OBJECT) DOMDocumentTypeImpl(*this, false, deep);
+        newNode = ::new (doc, DOMMemoryManager::DOCUMENT_TYPE_OBJECT) DOMDocumentTypeImpl(*this, false, deep);
     else
     {
         XMLMutexLock lock(sDocumentMutex);
-        newNode = new (sDocument, DOMMemoryManager::DOCUMENT_TYPE_OBJECT) DOMDocumentTypeImpl(*this, false, deep);
+        newNode = ::new (sDocument, DOMMemoryManager::DOCUMENT_TYPE_OBJECT) DOMDocumentTypeImpl(*this, false, deep);
     }
 
     fNode.callUserDataHandlers(DOMUserDataHandler::NODE_CLONED, this, newNode);
diff -ru xerces-c-3.1.1/src/xercesc/dom/impl/DOMDocumentTypeImpl.hpp xerces-c-3.1.1-gpdb/src/xercesc/dom/impl/DOMDocumentTypeImpl.hpp
--- xerces-c-3.1.1/src/xercesc/dom/impl/DOMDocumentTypeImpl.hpp	2008-03-26 01:06:57.000000000 -0700
+++ xerces-c-3.1.1-gpdb/src/xercesc/dom/impl/DOMDocumentTypeImpl.hpp	2015-11-30 12:00:29.146035567 -0800
@@ -44,7 +44,7 @@
 
 class DOMNamedNodeMapImpl;
 
-class CDOM_EXPORT DOMDocumentTypeImpl: public DOMDocumentType {
+class CDOM_EXPORT DOMDocumentTypeImpl: public DOMDocumentType, public XMemory {
 protected:
     DOMNodeImpl          fNode;
     DOMParentNode        fParent;
diff -ru xerces-c-3.1.1/src/xercesc/dom/impl/DOMImplementationImpl.cpp xerces-c-3.1.1-gpdb/src/xercesc/dom/impl/DOMImplementationImpl.cpp
--- xerces-c-3.1.1/src/xercesc/dom/impl/DOMImplementationImpl.cpp	2008-06-26 06:29:21.000000000 -0700
+++ xerces-c-3.1.1-gpdb/src/xercesc/dom/impl/DOMImplementationImpl.cpp	2015-11-30 12:00:29.148035552 -0800
@@ -152,7 +152,7 @@
     if(!XMLChar1_0::isValidName(qualifiedName))
         throw DOMException(DOMException::INVALID_CHARACTER_ERR, 0);
 
-    //to do: do we need to create with user's memorymanager???
+    // create with global memory manager
     DOMDocumentTypeImpl* docType = new DOMDocumentTypeImpl(0, qualifiedName, publicId, systemId, true);
     return docType;
 }
@@ -281,6 +281,7 @@
 
 DOMImplementationList* DOMImplementationImpl::getDOMImplementationList(const XMLCh* features) const
 {
+	// create with global memory manager
     DOMImplementationListImpl* list = new DOMImplementationListImpl;
     DOMImplementation* myImpl=getDOMImplementation(features);
     if(myImpl)
diff -ru xerces-c-3.1.1/src/xercesc/dom/impl/DOMNodeImpl.cpp xerces-c-3.1.1-gpdb/src/xercesc/dom/impl/DOMNodeImpl.cpp
--- xerces-c-3.1.1/src/xercesc/dom/impl/DOMNodeImpl.cpp	2008-07-02 09:00:35.000000000 -0700
+++ xerces-c-3.1.1-gpdb/src/xercesc/dom/impl/DOMNodeImpl.cpp	2015-11-30 12:00:29.150035538 -0800
@@ -36,6 +36,8 @@
 #include <xercesc/util/XMLUniDefs.hpp>
 #include <xercesc/util/PlatformUtils.hpp>
 #include <xercesc/util/XMLInitializer.hpp>
+#include <xercesc/util/XMemory.hpp>
+
 #include <stdio.h>
 #include <assert.h>
 
@@ -64,6 +66,7 @@
 
 void XMLInitializer::initializeDOMNodeListImpl()
 {
+	// create with global memory manager
     gEmptyNodeList = new DOMNodeListImpl(0);
 }
 
diff -ru xerces-c-3.1.1/src/xercesc/util/EncodingValidator.cpp xerces-c-3.1.1-gpdb/src/xercesc/util/EncodingValidator.cpp
--- xerces-c-3.1.1/src/xercesc/util/EncodingValidator.cpp	2008-03-10 07:10:09.000000000 -0700
+++ xerces-c-3.1.1-gpdb/src/xercesc/util/EncodingValidator.cpp	2015-11-30 12:00:29.176035348 -0800
@@ -32,6 +32,7 @@
 
 void XMLInitializer::initializeEncodingValidator()
 {
+	// create with global memory manager
     EncodingValidator::fInstance = new EncodingValidator();
 }
 
diff -ru xerces-c-3.1.1/src/xercesc/util/EncodingValidator.hpp xerces-c-3.1.1-gpdb/src/xercesc/util/EncodingValidator.hpp
--- xerces-c-3.1.1/src/xercesc/util/EncodingValidator.hpp	2008-03-10 07:10:09.000000000 -0700
+++ xerces-c-3.1.1-gpdb/src/xercesc/util/EncodingValidator.hpp	2015-11-30 12:00:29.177035341 -0800
@@ -34,7 +34,8 @@
   * encoding
   */
 
-class XMLUTIL_EXPORT EncodingValidator {
+class XMLUTIL_EXPORT EncodingValidator : public XMemory
+{
 
 public:
     // -----------------------------------------------------------------------
diff -ru xerces-c-3.1.1/src/xercesc/util/KVStringPair.hpp xerces-c-3.1.1-gpdb/src/xercesc/util/KVStringPair.hpp
--- xerces-c-3.1.1/src/xercesc/util/KVStringPair.hpp	2007-07-09 02:09:51.000000000 -0700
+++ xerces-c-3.1.1-gpdb/src/xercesc/util/KVStringPair.hpp	2015-11-30 12:00:29.179035326 -0800
@@ -82,12 +82,12 @@
     // -----------------------------------------------------------------------
     void setKey(const XMLCh* const newKey);
     void setValue(const XMLCh* const newValue);
-    void setKey
+    inline void setKey
     (
         const   XMLCh* const newKey
         , const XMLSize_t    newKeyLength
     );
-    void setValue
+    inline void setValue
     (
         const   XMLCh* const newValue
         , const XMLSize_t    newValueLength
diff -ru xerces-c-3.1.1/src/xercesc/util/XMLString.cpp xerces-c-3.1.1-gpdb/src/xercesc/util/XMLString.cpp
--- xerces-c-3.1.1/src/xercesc/util/XMLString.cpp	2010-01-20 00:45:02.000000000 -0800
+++ xerces-c-3.1.1-gpdb/src/xercesc/util/XMLString.cpp	2015-11-30 12:00:29.222035013 -0800
@@ -1364,7 +1364,7 @@
 
 
 XMLCh*
-XMLString::makeUName(const XMLCh* const pszURI, const XMLCh* const pszName)
+XMLString::makeUName(const XMLCh* const pszURI, const XMLCh* const pszName, MemoryManager* const  manager)
 {
     //
     //  If there is a URI, then format out the full name in the {uri}name
@@ -1374,8 +1374,8 @@
     const XMLSize_t uriLen = stringLen(pszURI);
     if (uriLen)
     {
-        pszRet = new XMLCh[uriLen + stringLen(pszName) + 3];
-
+        pszRet = (XMLCh*) manager->allocate((uriLen + stringLen(pszName) + 3) * sizeof(XMLCh)); // new XMLCh[uriLen + stringLen(pszName) + 3]
+        
         XMLCh szTmp[2];
         szTmp[1] = 0;
 
diff -ru xerces-c-3.1.1/src/xercesc/util/XMLString.hpp xerces-c-3.1.1-gpdb/src/xercesc/util/XMLString.hpp
--- xerces-c-3.1.1/src/xercesc/util/XMLString.hpp	2010-04-11 06:04:59.000000000 -0700
+++ xerces-c-3.1.1-gpdb/src/xercesc/util/XMLString.hpp	2015-11-30 12:00:29.222035013 -0800
@@ -1262,7 +1262,8 @@
     static XMLCh* makeUName
     (
         const   XMLCh* const    pszURI
-        , const XMLCh* const    pszName
+        , const XMLCh* const    pszName,
+        MemoryManager* const  manager = XMLPlatformUtils::fgMemoryManager
     );
 
     /**
Only in xerces-c-3.1.1/tests/src/xinclude/tests/xinclude: written
